client-app/pages/product.vue

```diff
@@ -369,11 +368,11 @@ watchEffect(async () => {
     await fetchRelatedProducts({ productId: productId.value, itemsPerPage: 30 });
   }
 
-  const recommendedProductsBlocks = (recommendedProductsSection.value?.blocks ?? []) as { model: string }[];
-  if (!recommendedProductsSection.value?.hidden && recommendedProductsSection.value?.blocks.length) {
+  const recommendedProductsBlocks = recommendedProductsSection.value?.blocks?.filter((block) => !block.model) ?? [];
```

Comment:
you sure new logic should not be tested by qa?

Reply:
Great that you paid attention to this line, because I made a mistake in condition. I fixed the condition and check recommendation blocks on myself - they work fine. Block with no model won't be showed anyway, because return no products, so I do filtration to avoid extra network request.